### DISCUSSION TITLE
Uninitialized constant `Rodauth::BCrypt` when creating a user in the console with `User.new` or `User.create`

### DISCUSSION THREAD

**Discussion by Samuelodan** [janko/rodauth-rails#289] [Q&A] [answered]
Hi there,
It's been a while. I hope you're well.
Please, I need help understanding this error. It seems BCrypt isn't required by default, which feels strange. If that's the case, how does creating users work with web requests?
I required "bcrypt" in the User model, and it started working in the console. Should BCrypt be required in my Gemfile?
Thanks.

```
irb(main):014> exp_user = User.new(first_name: "Demo", last_name: "Frank", email: "demofrank@example.com", password: "password", status: "verified")
/Users/samuelodan/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rodauth-2.33.0/lib/rodauth/features/login_password_requirements_base.rb:79:in `password_hash': uninitialized constant Rodauth::BCrypt (NameError)

      BCrypt::Password.create(password, :cost=>password_hash_cost)
            ^^^^^^^^^^
```

**Comment by janko** [comment, marked as answer]
See https://github.com/janko/rodauth-rails/issues/279#issuecomment-1961647540.

**Reply by Samuelodan** [reply]
Ah, that fixes it. Thank you.
